src/theory/quantifiers/sygus/cegis.cpp

```diff
@@ -198,6 +198,34 @@ bool Cegis::addEvalLemmas(const std::vector<Node>& candidates,
   return addedEvalLemmas;
 }
 
+Node Cegis::getRefinementLemmaFormula()
+{
+  std::vector<Node> conj;
+  conj.insert(
+      conj.end(), d_refinement_lemmas.begin(), d_refinement_lemmas.end());
+  // get the propagated values
+  for (unsigned i = 0, nprops = d_rl_eval_hds.size(); i < nprops; i++)
+  {
+    conj.push_back(d_rl_eval_hds[i].eqNode(d_rl_vals[i]));
+  }
+  // make the formula
+  NodeManager* nm = NodeManager::currentNM();
+  Node ret;
+  if (conj.empty())
+  {
+    ret = nm->mkConst(true);
+  }
+  else if (conj.size() == 1)
+  {
+    ret = conj[0];
+  }
+  else
+  {
+    ret = nm->mkNode(AND, conj);
+  }
```

Comment:
I suppose you'd not like to inline this to ret = cond1? ... : cond2? ... : ...?

Reply:
It gets a little ugly if its highly nested, I'll merge the last 2 blocks though.